wcfsetup/install/files/lib/action/FacebookAuthAction.class.php

```diff
@@ -68,7 +68,7 @@ public function execute() {
 			}
 			
 			// validate state, validation of state is executed after fetching the access_token to invalidate 'code'
-			if (!isset($_GET['state']) || $_GET['state'] != WCF::getSession()->getVar('__facebookInit')) throw new IllegalLinkException();
+			if (!isset($_GET['state']) || !\hash_equals($_GET['state'], WCF::getSession()->getVar('__facebookInit'))) throw new IllegalLinkException();
```

Comment:
You need to switch the parameter order. The first parameter is defined to be the known string, the second is defined to be the user string.

It does not matter with the *current* implementation of `hash_equals` within the C source of PHP, but it might matter in case of later changes.

```suggestion
			if (!isset($_GET['state']) || !\hash_equals(WCF::getSession()->getVar('__facebookInit'), $_GET['state'])) throw new IllegalLinkException();
```